taskcluster/scripts/podman-worker.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+import subprocess
+import sys
+import tempfile
+import urllib.parse
+import urllib.request
+
+def get_task_payload():
+    rootUrl = os.environ.get("TASKCLUSTER_ROOT_URL", "https://firefox-ci-tc.services.mozilla.com")
+    taskUrl = urllib.parse.urljoin(rootUrl, f'api/queue/v1/task/{os.environ.get("TASK_ID")}')
+    with urllib.request.urlopen(taskUrl) as req:
+        return json.load(req)["payload"]
+
+# Load the podman image and return its description in JSON
+def load_podman_image(path):
+    # Load the image and try to grab the tag.
+    tag = None
+    args = ['podman', 'load', '-i', path]
+    with subprocess.Popen(args, stdout=subprocess.PIPE, stderr=sys.stderr, text=True) as proc:
+        for line in proc.stdout:
+            print(f'{line}') # Foreward to stdout
+            if line.startswith('Loaded image'):
+                i = line.index(':')
+                if i > 0:
+                    tag = line[i+1:].strip()
+
+    # Describe the podman image. 
+    x = subprocess.run(['podman', 'image', 'inspect', tag], stdout=subprocess.PIPE, stderr=sys.stderr)
+    return json.loads(x.stdout)[0]
+
+moz_fetches_dir = os.path.realpath(os.environ.get("MOZ_FETCHES_DIR"))
+
+# Load the podman image.
+image_filename = os.path.join(moz_fetches_dir, 'image.tar.zst')
+print(f"Loading podman image from: {image_filename}")
+sys.stdout.flush()
+image_data = load_podman_image(image_filename)
+image_config = image_data["Config"]
+
+# Prepare the task arguments
+worker_env = get_task_payload()["env"]
+worker_args = ['podman', 'run', '--rm', '--privileged']
```

Comment:
Just noting that using `--privileged` should be fine because we aren't running this as root, and also `generic-worker` creates a new user for each task. So unlike Docker which forwards commands to a daemon that does run as root, using this flag with podman should be safe from a task isolation standpoint.